### DISCUSSION TITLE
How to reference an old value in an update write statement?

### DISCUSSION THREAD

**Discussion by TheSHEEEP** [simolus3/drift#3357] [Q&A] [answered]
I'd like to do something like the following:

```
await(db.select(db.homes)..where((tbl) => tbl.sortedIndex.isBiggerThanValue(oldIndex)))
      .write(
        HomesCompanion(
          sortedIndex: Value(old_sortedIndex_value - const Constant(1)),
        )
    );
```

But I have no idea how to get that old_sortedIndex_value. And in the documentation, I cannot find anything about referencing a current value as you would in a simple SQL statement like "UPDATE ... WHERE ... SET sortedIndex = sortedIndex - 1".

Currently, all I can come up with is selecting the entries and then manually updating each of them in a second step, but that is of course wasting performance. Or do a fully custom string query.   
But there must be a way using Drift functions, no?

**Comment by simolus3** [comment, marked as answer]
The current documentation on the website for this is not great, and I'll improve that. But companions have a `.custom` static method on them that can be used to build custom SQL expressions into updates. So you could write your update as

```dart
  await (db.update(db.homes)..where((tbl) => tbl.sortedIndex.isBiggerThanValue(oldIndex)))
      .write(HomesCompanion.custom(sortedIndex: db.homes.sortedIndex - const Constant(1)));
```

**Reply by TheSHEEEP** [reply]
That did the trick, thank you.

And yeah, I agree, some section about how to reference existing values in a write would definitely be very helpful.
Or a section about the companions and what they can do and then reference that in the "Writes" section.
